src/NuGet.Core/NuGet.Versioning/VersionFormatter.cs

```diff
@@ -54,7 +54,7 @@ public object GetFormat(Type formatType)
         {
             if (formatType == typeof(ICustomFormatter)
                 || formatType == typeof(NuGetVersion)
-                || formatType == typeof(SemanticVersion))
+                || typeof(SemanticVersion).IsAssignableFrom(formatType))
```

Comment:
```suggestion
                || typeof(SemanticVersion).IsAssignableFrom(formatType))
```

Reply:
I put together a benchmark. Here are the results

|                 Method |       Implementation |      Mean |     Error |    StdDev |    Median |
|----------------------- |--------------------- |----------:|----------:|----------:|----------:|
| CompareSemanticVersion |             Fallback |  4.907 ns | 0.1339 ns | 0.1252 ns |  4.875 ns |
|    CompareNuGetVersion |             Fallback |  5.226 ns | 0.1459 ns | 0.2478 ns |  5.179 ns |
|   CompareCustomVersion |             Fallback | 24.932 ns | 0.5226 ns | 0.9816 ns | 24.726 ns |
|          CompareObject |             Fallback | 26.623 ns | 2.5718 ns | 7.5829 ns | 22.854 ns |
| CompareSemanticVersion | Singl(...)eFrom [22] |  6.285 ns | 0.6062 ns | 1.7682 ns |  6.419 ns |
|    CompareNuGetVersion | Singl(...)eFrom [22] | 13.948 ns | 0.3181 ns | 0.6279 ns | 13.804 ns |
|   CompareCustomVersion | Singl(...)eFrom [22] | 13.733 ns | 0.3106 ns | 0.4147 ns | 13.701 ns |
|          CompareObject | Singl(...)eFrom [22] | 17.396 ns | 0.3803 ns | 0.3735 ns | 17.406 ns |

In absolute terms, they're all very quick, the difference wouldn't be noticeable to humans unless a process does it an absurd number of times. Although technically removing the two `==` comparisons, NuGet's primary use case (visual studio, dotnet cli) would become marginally slower. However it would be faster for customers who do implement their own version by extending NuGet's version classes.  Anyway, our team's primary responsibility is the VS & dotnet CLI experience, so I appreciate the suggestion, but I'm not going to create a new PR to merge it.

<details>
<summary>code</summary>
<div>

```cs
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using NuGet.Versioning;

public class Program
{
    static void Main() => BenchmarkRunner.Run<Program>();

    public enum Method
    {
        Fallback,
        SingleIsAssignableFrom
    }

    [ParamsAllValues]
    public Method Implementation { get; set; }

    private Func<Type, bool>? _compare;
    private SemanticVersion _semanticVersion = SemanticVersion.Parse("1.2.3");
    private NuGetVersion _nugetVersion = NuGetVersion.Parse("1.2.3");
    private CustomVersion _customVersion = new CustomVersion(1, 2, 3);

    [GlobalSetup]
    public void GlobalSetup()
    {
        _compare = Implementation switch
        {
            Method.Fallback =>
                type =>
                    type == typeof(SemanticVersion)
                    || type == typeof(NuGetVersion)
                    || typeof(SemanticVersion).IsAssignableFrom(type),
            Method.SingleIsAssignableFrom =>
                type =>
                    typeof(SemanticVersion).IsAssignableFrom(type),
            _ => throw new NotImplementedException()
        };
    }

    [Benchmark]
    public bool CompareSemanticVersion()
    {
        return _compare!(typeof(SemanticVersion));
    }

    [Benchmark]
    public bool CompareNuGetVersion()
    {
        return _compare!(typeof(NuGetVersion));
    }

    [Benchmark]
    public bool CompareCustomVersion()
    {
        return _compare!(typeof(CustomVersion));
    }

    [Benchmark]
    public bool CompareObject()
    {
        return _compare!(typeof(object));
    }

    private class CustomVersion : NuGetVersion
    {
        public CustomVersion(int major, int minor, int patch)
            : base(major, minor, patch)
        {
        }
    }
}
```

</div>
</details>